ext/pdo_firebird/firebird_statement.c

```diff
@@ -758,7 +756,7 @@ static int pdo_firebird_stmt_param_hook(pdo_stmt_t *stmt, struct pdo_bound_param
 					ZEND_FALLTHROUGH;
 				case IS_NULL:
 					/* complain if this field doesn't allow NULL values */
-					if (~var->sqltype & 1) {
+					if (UNEXPECTED(~var->sqltype & 1)) {
```

Comment:
Globally looks correct to me, maybe this one (and the one above) kind of look a bit much. Not sure if there is a real gain but that s nitpicking :)